arrow-array/src/array/primitive_array.rs

```diff
@@ -233,7 +233,8 @@ pub type Decimal256Array = PrimitiveArray<Decimal256Type>;
 /// static-typed nature of rust types ([`ArrowNativeType`]) for all types that implement [`ArrowNativeType`].
 pub trait ArrowPrimitiveType: 'static {
     /// Corresponding Rust native type for the primitive type.
-    type Native: ArrowNativeType;
+    // FIXME: this may affect other code, is this correct?
+    type Native: ArrowNativeType + ArrowNativeTypeOp;
```

Comment:
```suggestion
    type Native: ArrowNativeTypeOp;
```

`ArrowNativeType` is implied by `ArrowNativeTypeOp`.

I think this change is fine, but tagging @viirya who added `ArrowNativeTypeOp` originally

Reply:
yes, It is indeed better to write this way